kyo-combinators/shared/src/main/scala/kyo/Combinators.scala

```diff
@@ -136,44 +136,43 @@ end extension
 
 extension [A, S, E](effect: A < (Abort[E] & S))
     def handleAbort(
-        using
-        ct: ClassTag[E],
-        tag: Tag[E],
-        flat: Flat[A]
+        using flat: Flat[A]
     ): Result[E, A] < S =
-        Abort.run[E](effect)
+        Abort.run[Any](effect).asInstanceOf[Result[E, A] < S]
```

Comment:
@fwbrasil what do you think of catching all errors? I see that `Tag` doesn't support union types, but there should be some way to handle all composed error types at one time.

The danger in this approach is that it will end up handling "hidden" `Abort` effect (e.g., opaque type alias on `Abort[X]` for some failure `X`).

Reply:
The previous behavior allowing type unions was unsound because the `ClassTag` gets inferred as `Any` in those cases, essentially handling any `Abort`, not only the ones in the type union. The `Tag` was added to `Abort.run` as a way to prevent this scenario. We might be able to eventually lift this limitation with the new tag implementation I'm planning to work on.  

I suggest keeping it restricted here as well to avoid issues.